sdk/eventhub/azure-eventhub/azure/eventhub/aio/_client_base_async.py

```diff
@@ -107,12 +107,15 @@ def __init__(
         self,
         fully_qualified_namespace: str,
         eventhub_name: str,
-        credential: Union["AsyncTokenCredential", AzureSasCredential],
+        credential: Union["AsyncTokenCredential", AzureSasCredential, AzureNamedKeyCredential],
         **kwargs: Any
     ) -> None:
         self._loop = kwargs.pop("loop", None)
         if isinstance(credential, AzureSasCredential):
             self._credential = AzureSasTokenCredentialAsync(credential) # type: ignore
+        elif isinstance(credential, AzureNamedKeyCredential):
+            name, key = credential.named_key
+            self._credential = EventHubSharedKeyCredential(name, key) # type: ignore
```

Comment:
same comment, I think the name-key pair should be extracted from the credential in EventHubSharedKeyCredential.get_token(self, *scopes, **kwargs):.

or we could introduce a separate class for this AzureNamedKeyCredential if we don't want to "pollute" the public EventHubSharedKeyCredential class?